src/common/configs/web-visualization-configuration-factory.ts

```diff
@@ -65,17 +61,67 @@ export class WebVisualizationConfigurationFactory implements VisualizationConfig
         return configuration;
     }
 
-    public getChromeCommandToVisualizationTypeMap(): DictionaryStringTo<VisualizationType> {
-        const map: DictionaryStringTo<VisualizationType> = {};
+    public forEachConfig(callback: ForEachConfigCallback): void {
+        Object.keys(this.configurationByType).map(type => {
+            callback(this.configurationByType[type], Number(type));
+        });
+
+        this.fullAssessmentProvider.all().map(assessment => {
+            const testConfig = this.buildAssessmentConfiguration(assessment, TestMode.Assessments);
 
-        const types = EnumHelper.getNumericValues<VisualizationType>(VisualizationType);
+            assessment.requirements.map(requirementConfig => {
+                callback(testConfig, assessment.visualizationType, requirementConfig);
+            });
+        });
+
+        this.mediumPassProvider?.all().map(assessment => {
+            const testConfig = this.buildAssessmentConfiguration(assessment, TestMode.MediumPass);
+
+            assessment.requirements.map(requirementConfig => {
+                callback(testConfig, assessment.visualizationType, requirementConfig);
+            });
+        });
+    }
 
-        each(types, visualizationType => {
-            const configuration = this.configurationByType[visualizationType];
+    private buildAssessmentConfiguration(
+        assessment: Assessment,
+        testMode: TestMode,
+    ): VisualizationConfiguration {
+        const config = assessment.getVisualizationConfiguration();
 
-            if (configuration && configuration.chromeCommand != null) {
-                map[configuration.chromeCommand] = visualizationType;
+        const getIdentifier = step => {
```

Comment:
```suggestion
        const getIdentifier = testStep: string => {
```

Explicit types increase readability, changing variable name to match declaration in `VisualizationConfiguration`.